Added support for game controllers

LukePH opened the pull request:
-Added support for game controllers
  -Will take over CPU players on button press. (TODO: Missing feedback on take over)
  -Supports hotswapping of new controllers
-Fix player's healthbar drawing too large with extra health, and not being removed when player is dead
-Brightened up the walls a bit

LukePH committed:
-Added support for game controllers
  -Will take over CPU players on button press. (TODO: Missing feedback on take over)
  -Supports hotswapping of new controllers
-Fix player's healthbar drawing too large with extra health, and not being removed when player is dead
-Brightened up the walls a bit

VenKamikaze commented:
Thanks for this!

I've merged this to sdl2_ongoing for now, as I'm hitting an occasional segfault. It's probably not related to the controller code itself - I suspect I've only hit it now that I've tested playing a 2 player game.

LukePH commented:
Interesting, not sure were that would be coming from,

if you run the game with "gdb -ex run ./BitRiot" when it crashes you can get a stack by typing: 'backtrace' you won't get line numbers unless it's compiled with debug symbols turned on but should help unless it's a memory corruption bug.

LukePH committed:
Fix what looks like a possible crash in input code after at least one player has died. (I haven't seem a crash from myself, but looks likely.)

I missed the fact that p_players is not zeroed after a player is deleted, and instead a second array is used the handle the dangling pointers.

An array like this isn't really normal or safe. Would be better to null the pointers, or use a smart pointer or something like this http://corrodinggames.com/forums/viewtopic.php?f=2&t=12 :p
